projects/aws/bottlerocket-bootstrap/pkg/providers/snow/system/reboot.go

```diff
@@ -0,0 +1,43 @@
+package system
+
+import (
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+
+	"github.com/eks-anywhere-build-tooling/aws/bottlerocket-bootstrap/pkg/files"
+	"github.com/eks-anywhere-build-tooling/aws/bottlerocket-bootstrap/pkg/utils"
+)
+
+const (
+	bootstrapContainerPath = "/.bottlerocket/bootstrap-containers/current"
+)
+
+func rebootInstanceIfNeeded() error {
+	rebootedPath := filepath.Join(bootstrapContainerPath, "rebooted")
+
+	// Skip reboot when instance is already rebooted once
+	if files.PathExists(rebootedPath) {
+		return nil
+	}
+
+	if err := reboot(); err != nil {
+		return err
+	}
+
+	// Write rebooted file to make sure reboot only happens once
+	if _, err := os.Create(rebootedPath); err != nil {
```

Comment:
this should be moved to before `if err := reboot();`, otherwise I think it won't be executed?